--- qabstractitemmodel_old-57fd823ff2b59048618b9c01ec4cf329baa40234.h	2013-06-28 12:38:44.000000000 +1000
+++ qabstractitemmodel_new-3d54594a4a7918f148f0d535198d963cdc4c4aca.h	2013-07-02 07:31:54.000000000 +1000
@@ -98,6 +98,9 @@
 
 class QPersistentModelIndexData;
 
+// qHash is a friend, but we can't use default arguments for friends (§8.3.6.4)
+uint qHash(const QPersistentModelIndex &index, uint seed = 0);
+
 class Q_CORE_EXPORT QPersistentModelIndex
 {
 public:
@@ -128,7 +131,7 @@
     bool isValid() const;
 private:
     QPersistentModelIndexData *d;
-    friend uint qHash(const QPersistentModelIndex &, uint seed = 0);
+    friend uint qHash(const QPersistentModelIndex &, uint seed);
 #ifndef QT_NO_DEBUG_STREAM
     friend Q_CORE_EXPORT QDebug operator<<(QDebug, const QPersistentModelIndex &);
 #endif
