--- qhostaddress_old-238ca251c911101feef68cdca102bbdf1335a4d5.h	2013-06-28 12:38:44.000000000 +1000
+++ qhostaddress_new-2cacda806984da7f615d3bcccd451c5229bb8760.h	2013-07-02 07:31:54.000000000 +1000
@@ -64,6 +64,10 @@
 
 typedef QIPv6Address Q_IPV6ADDR;
 
+class QHostAddress;
+// qHash is a friend, but we can't use default arguments for friends (§8.3.6.4)
+Q_NETWORK_EXPORT uint qHash(const QHostAddress &key, uint seed = 0);
+
 class Q_NETWORK_EXPORT QHostAddress
 {
 public:
@@ -121,7 +125,7 @@
 
     static QPair<QHostAddress, int> parseSubnet(const QString &subnet);
 
-    friend Q_NETWORK_EXPORT uint qHash(const QHostAddress &key, uint seed = 0);
+    friend Q_NETWORK_EXPORT uint qHash(const QHostAddress &key, uint seed);
 protected:
     QScopedPointer<QHostAddressPrivate> d;
 };
